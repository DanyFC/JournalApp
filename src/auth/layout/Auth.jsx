import { Grid, Typography } from '@mui/material'

const Auth = ({ children, title = '' }) => {
  return (
    <Grid
      alignItems='center'
      container
      direction='column'
      justifyContent='center'
      spacing={0}
      sx={{ minHeight: '100vh', backgroundColor: 'primary.main', padding: 4 }}
    >
      <Grid
        className='box-shadow animate__animated animate__fadeInDown'
        item
        sx={{ backgroundColor: 'white', padding: 3, borderRadius: 2, width: { sm: 450 } }}
        xs={3}
      >
        <Typography variant='h5' sx={{ mb: 1 }}>{title}</Typography>
        {children}
      </Grid>
    </Grid>
  )
}
export default Auth