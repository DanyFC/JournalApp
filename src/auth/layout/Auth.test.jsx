/**
 * @jest-environment jsdom
*/
import { render, screen } from '@testing-library/react'
import Auth from './Auth'

/* eslint-disable no-undef */
describe('Tests in <Auth />', () => {

  const layoutContent = {
    title: 'Title layout',
    children: <h1>Children Layout</h1>
  }

  test('Can be match with snapshot.', () => {
    const { container } = render(
      <Auth title={layoutContent.title}>
        {layoutContent.children}
      </Auth>
    )

    expect(container).toMatchSnapshot()
  })

  test('Should show the title and the children.', () => {
    render(
      <Auth title={layoutContent.title}>
        {layoutContent.children}
      </Auth>
    )

    expect(screen.getByText(layoutContent.title)).toBeTruthy()
    expect(screen.getByText(layoutContent.children.props.children)).toBeTruthy()
  })
})
/* eslint-enable no-undef */
