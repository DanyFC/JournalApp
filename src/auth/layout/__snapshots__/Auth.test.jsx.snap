// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Auth /> Can be match with snapshot. 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-ro1yt6-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 box-shadow animate__animated animate__fadeInDown css-1ar3nvi-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-1jcm1rz-MuiTypography-root"
      >
        Title layout
      </h5>
      <h1>
        Children Layout
      </h1>
    </div>
  </div>
</div>
`;
